pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -454,9 +456,28 @@ class Utils {
     @Restricted(NoExternalUse.class)
     static <T> T instantiateDescribable(Class<T> c, Map<String, ?> args) {
         DescribableModel<T> model = new DescribableModel<>(c)
+        // Special case for JENKINS-63499.
+        if (model != null && model.type == PasswordParameterDefinition.class && model.getParameter("defaultValueAsSecret") != null) {
+            args = copyMapReplacingEntry(args, "defaultValue", "defaultValueAsSecret", String.class, { s -> Secret.fromString(s) })
+        }
         return model?.instantiate(args)
     }
 
+    /**
+     * Copy a map, replacing the entry with the specified key if it matches the specified type.
+     */
+    public static <T> Map<String, Object> copyMapReplacingEntry(Map<String, ?> map, String oldKey, String newKey, Class<T> requiredValueType, Function<T, Object> replacer) {
+        Map<String, Object> newMap = new TreeMap<>()
+        for (Map.Entry<String, ?> entry : map.entrySet()) {
+            if (entry.getKey().equals(oldKey) && requiredValueType.isInstance(entry.getValue())) {
+                newMap.put(newKey, replacer.apply(requiredValueType.cast(entry.getValue())))
+            } else {
+                newMap.put(entry.getKey(), entry.getValue())
+            }
+        }
+        newMap
```

Comment:
How come there is no `return` keyword here?

Reply:
Because this is Groovy 😢 . If I had some spare time on my hands I would like to convert everything in `src/main/groovy` to Java for better IDE support and to avoid confusion (see also a few lines up in the other method where I have to pass a closure to a method expecting a `Function`, since Groovy doesn't support Java 8 lambdas directly, instead it coerces the closure to the interface type)

Reply:
My bad, I totally missed that!